features/timeseries/src/main/java/org/opennms/netmgt/timeseries/integration/persistence/TimeseriesPersisterFactory.java

```diff
@@ -47,13 +52,19 @@
 public class TimeseriesPersisterFactory implements PersisterFactory {
 
     private final TimeseriesWriter timeseriesWriter;
-
     private final Context context;
+    private final MetaTagDataLoader metaTagDataLoader;
+    private final Map<ResourcePath, Map<String, String>> metaCache;
 
     @Inject
-    public TimeseriesPersisterFactory(Context context, TimeseriesWriter timeseriesWriter) {
+    public TimeseriesPersisterFactory(final Context context, final TimeseriesWriter timeseriesWriter, final MetaTagDataLoader metaTagDataLoader) {
         this.context = Objects.requireNonNull(context);
         this.timeseriesWriter = timeseriesWriter;
+        this.metaTagDataLoader = metaTagDataLoader;
+        Cache<ResourcePath, Map<String, String>> cache = CacheBuilder // TODO Patrick: define with Jesse the eviction policy
```

Comment:
@j-white how often do we expect the Resource Meta Data to change / what should our eviction policy be?

Reply:
Let's use expireAfterWrite with a default of 5m. (And make it configurable.)